scripts/mellanox_buffer_migrator.py

```diff
@@ -834,3 +835,193 @@ def mlnx_flush_new_buffer_configuration(self):
+            ingress_lossy_pg_zero_profile = {
+                "pool":"[BUFFER_POOL|ingress_zero_pool]",
+                "size":"0",
+                "static_th":"0"
+            }
+            lossy_pg_item = {'profile': '[BUFFER_PROFILE|ingress_lossy_pg_zero_profile]'}
+
+            ingress_lossless_zero_profile = {
+                "pool":"[BUFFER_POOL|ingress_lossless_pool]",
+                "size":"0",
+                "dynamic_th":"-8"
+            }
+
+            if single_pool:
+                ingress_profile_list_item = {'profile_list': '[BUFFER_PROFILE|ingress_lossless_zero_profile]'}
+            else:
+                ingress_lossy_zero_profile = {
+                    "pool":"[BUFFER_POOL|ingress_lossy_pool]",
+                    "size":"0",
+                    "dynamic_th":"-8"
+                }
+                ingress_profile_list_item = {'profile_list': '[BUFFER_PROFILE|ingress_lossless_zero_profile],[BUFFER_PROFILE|ingress_lossy_zero_profile]'}
+
+            egress_lossless_zero_profile = {
+                "pool":"[BUFFER_POOL|egress_lossless_pool]",
+                "size":"0",
+                "dynamic_th":"-8"
+            }
+            lossless_queue_item = {'profile': '[BUFFER_PROFILE|egress_lossless_zero_profile]'}
+
+            egress_lossy_zero_profile = {
+                "pool":"[BUFFER_POOL|egress_lossy_pool]",
+                "size":"0",
+                "dynamic_th":"-8"
+            }
+            lossy_queue_item = {'profile': '[BUFFER_PROFILE|egress_lossy_zero_profile]'}
+            egress_profile_list_item = {'profile_list': '[BUFFER_PROFILE|egress_lossless_zero_profile],[BUFFER_PROFILE|egress_lossy_zero_profile]'}
+
+            queue_items_to_apply = {'0-2': {'profile': '[BUFFER_PROFILE|egress_lossy_zero_profile]'},
+                                    '3-4': {'profile': '[BUFFER_PROFILE|egress_lossless_zero_profile]'},
+                                    '5-6': {'profile': '[BUFFER_PROFILE|egress_lossy_zero_profile]'}}
+
+            pools_to_insert = {'ingress_zero_pool': ingress_zero_pool}
+            profiles_to_insert = {'ingress_lossy_pg_zero_profile': ingress_lossy_pg_zero_profile,
+                                  'ingress_lossless_zero_profile': ingress_lossless_zero_profile,
+                                  'egress_lossless_zero_profile': egress_lossless_zero_profile,
+                                  'egress_lossy_zero_profile': egress_lossy_zero_profile}
+            if not single_pool:
+                profiles_to_insert['ingress_lossy_zero_profile'] = ingress_lossy_zero_profile
+
+        lossless_profile_pattern = '\[BUFFER_PROFILE\|pg_lossless_([1-9][0-9]*000)_([1-9][0-9]*m)_profile\]'
+        zero_item_count = 0
+        for port_name, port_info in port_table.items():
+            if port_info.get('admin_status') == 'up':
+                # Handles admin down ports only
+                continue
+
+            if lossy_pg_item:
```

Comment:
If a candidate of BUFFER_PG table has been generated, we will check whether the BUFFER_PG table aligns with the default value and apply profiles if yes.